docs/sphinx/user_guide/feature/list.rst

```diff
@@ -0,0 +1,256 @@
+
+.. code-block:: cpp 
+
+    using list1 = camp::list<int&, int&>;
+
+    using list2 = camp::transform<std::remove_reference, list1>;
+    // list2 is of type camp::list<int, int>;
+
+Operating on lists with the ``Accumulate`` construct
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``accumulate`` construct can be used to apply a given operation to a list. ``Accumulate`` 
+takes an operation, an initial value, and a list. It applies the operation across the list, starting with 
+the initial value.
+
+.. code-block:: cpp 
+
+    using myNewList = accumulate<append, list<>, list<int, float, double>>; 
+    // myNewList is of type list<int, float, double>;
+
+Cartesian products of lists (an application of accumulate)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Camp provides a method to evaluate the cartesian product of two lists. The ``cartesian_product`` method is simply an 
+application of the accumulate method. 
+
+.. code-block:: cpp 
+
+    struct a;
+    struct b;
+    struct c;
+    struct d;
+    struct e;
+
+    using listA = list<a, b>;
+    using listB = list<c, d, e>
+
+    using prod = cartesian_product<listA, listB>;
+    // prod is of type list<list<a, c>,
+    //                      list<a, d>,
+    //                      list<a, e>,
+    //                      list<b, c>,
+    //                      list<b, d>,
+    //                      list<b, e>>
+
+Finding the first index of a type within a list
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``index_of`` method can be used on a list to find the first index in the list where
+a given type appears. If the type is not found in a list, ``nil`` is returned. 
+
+.. code-block:: cpp 
+
+    using myList = list<int, double, char, float>
+
+    using firstChar = index_of<char, myList>::type
+    // fistChar is num<2>
+    using firstBool = index_of<bool, myList>::type
+    // firstBool is nil
+
+
+Filtering Types with ``filter``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Camp provides a way to filter a list such that only the desired types remain. 
+
+.. code-block:: cpp
+
+    using myList = list<int, float*, double, short*>;
+
+    using ptrsOnly = filter<std::is_pointer, myList>;
+    // ptrsOnly is of type list<float*, short*>
+
+------------------------
+Using Lists to make Maps
+------------------------
+
+Camp provides a ``map.hpp`` header which can be combined with associative lists 
+to create a map-type structure. Consider the following: 
+
+.. code-block:: cpp 
+
+    using myMap = list<list<int, num<0>>, list<char, num<1>>>;
+
+Using the ``at_key`` method, we can do a lookup on the maps "keys" to access its "values". This works 
+because the lists  act as key value pairs. Here is an example of a lookup using the ``at_key`` method: 
+
+.. code-block:: cpp  
+
+    using myMap = list<list<int, num<0>>, list<char, num<1>>>;
+
+    using val = at_key<myList, int>;
+
+    // val is num<0>
+
```

Comment:
Camp also has an 'at_v' method to pull actual values out of a list in addition to types. We use this in places in RAJA tests:

https://github.com/LLNL/RAJA/blob/develop/test/functional/kernel/single-loop-tile-icount-tcount/test-kernel-single-loop-tile-count.cpp.in#L33

Reply:
Okie, I added an example of how `at_v` can be used to retrieve values. Thanks for explaining that better!